### PR TITLE
fix(android,remote_config): ensureInitialized() task should ignore return value (fixes #5222)

### LINKED ISSUE #5222: 🐛 [remote_config] java.lang.IllegalArgumentException: Unsupported value: com.google.firebase.remoteconfig.internal.FirebaseRemoteConfigInfoImpl
## Bug report

After upgrading my app to use Flutter 2.0, and the latest firebase packages (firebase_core 1.0.0, firebase_remote_config 0.9.0-dev.0). My app starts initialising but the instantly crashes with the following error: 

```
D/AndroidRuntime( 7514): Shutting down VM
E/AndroidRuntime( 7514): FATAL EXCEPTION: main
E/AndroidRuntime( 7514): Process: com.rocketware.onescene, PID: 7514
E/AndroidRuntime( 7514): java.lang.IllegalArgumentException: Unsupported value: com.google.firebase.remoteconfig.internal.FirebaseRemoteConfigInfoImpl@39b686
E/AndroidRuntime( 7514): 	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:276)
E/AndroidRuntime( 7514): 	at io.flutter.plugin.common.StandardMethodCodec.encodeSuccessEnvelope(StandardMethodCodec.java:59)
E/AndroidRuntime( 7514): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:238)
E/AndroidRuntime( 7514): 	at io.flutter.plugins.firebase.firebaseremoteconfig.FirebaseRemoteConfigPlugin.lambda$onMethodCall$2(FirebaseRemoteConfigPlugin.java:167)
E/AndroidRuntime( 7514): 	at io.flutter.plugins.firebase.firebaseremoteconfig.-$$Lambda$FirebaseRemoteConfigPlugin$G3sJbc89zgSFW2R5fKq_9HoTj0c.onComplete(Unknown Source:2)
E/AndroidRuntime( 7514): 	at com.google.android.gms.tasks.zzi.run(com.google.android.gms:play-services-tasks@@17.0.2:4)
E/AndroidRuntime( 7514): 	at android.os.Handler.handleCallback(Handler.java:938)
E/AndroidRuntime( 7514): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 7514): 	at com.google.android.gms.internal.tasks.zzb.dispatchMessage(com.google.android.gms:play-services-tasks@@17.0.2:6)
E/AndroidRuntime( 7514): 	at android.os.Looper.loop(Looper.java:223)
E/AndroidRuntime( 7514): 	at android.app.ActivityThread.main(ActivityThread.java:7660)
E/AndroidRuntime( 7514): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 7514): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
E/AndroidRuntime( 7514): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
I/Process ( 7514): Sending signal. PID: 7514 SIG: 9
Lost connection to device.
Exited (sigterm)

```

---

## Additional context

I have included the full output I get when launching my app as there are lots of mentions of deprecated API's. Here is a sample:
```
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_remote_config-0.9.0-dev.0\android\src\main\java\io\flutter\plugins\firebase\firebaseremoteconfig\FirebaseRemoteConfigPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
```
<details><summary>Click To Expand</summary>

```
Running "flutter pub get" in os_app...
Launching lib\main.dart on Pixel 5 in debug mode...
 lib\main.dart
D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\shared_preferences-0.5.7+3\android\src\main\java\io\flutter\plugins\sharedpreferences\SharedPreferencesPlugin.java:18: warning: [deprecation] Registrar in PluginRegistry has been deprecated
  public static void registerWith(PluginRegistry.Registrar registrar) {
                                                ^
1 warning
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_analytics-5.0.15\android\src\main\java\io\flutter\plugins\firebaseanalytics\FirebaseAnalyticsPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_core-1.0.0\android\src\main\java\io\flutter\plugins\firebase\core\FlutterFirebaseCorePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\cloud_functions-1.0.0\android\src\main\java\io\flutter\plugins\firebase\functions\FlutterFirebaseFunctionsPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_storage-8.0.0\android\src\main\java\io\flutter\plugins\firebase\storage\FlutterFirebaseStoragePlugin.java:38: warning: [deprecation] Registrar in PluginRegistry has been deprecated
  public static void registerWith(PluginRegistry.Registrar registrar) {

                                                ^
1 warning
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_auth-1.0.0\android\src\main\java\io\flutter\plugins\firebase\auth\FlutterFirebaseAuthPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\firebase_remote_config-0.9.0-dev.0\android\src\main\java\io\flutter\plugins\firebase\firebaseremoteconfig\FirebaseRemoteConfigPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\emoji_picker-0.1.0\android\src\main\java\com\jeffg\emoji_picker\EmojiPickerPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\cloud_firestore-1.0.0\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestorePlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\cloud_firestore-1.0.0\android\src\main\java\io\flutter\plugins\firebase\firestore\streamhandler\TransactionStreamHandler.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\streams_channel-0.3.0\android\src\main\java\app\loup\streams_channel\StreamsChannelPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\google_sign_in-4.5.9\android\src\main\java\io\flutter\plugins\googlesignin\GoogleSignInPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\image_cropper-1.3.1\android\src\main\java\vn\hunghd\flutter\plugins\imagecropper\ImageCropperPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: D:\dev\sdks\flutter\.pub-cache\hosted\pub.dartlang.org\path_provider-1.6.1\android\src\main\java\io\flutter\plugins\pathprovider\PathProviderPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
√ Built build\app\outputs\flutter-apk\app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:55916/szdZKGkl0tY=/ws
I/flutter ( 7514): 14:24 Firebase/FINE: Init firebase...
D/FRCPlugin( 7514): Sending fetchTimeout: 60
I/flutter ( 7514): 14:98 ReduxLogging/INFO: {Action: Instance of 'InitAction', State: StoreState, ts: 2021-03-04 16:49:14.096931}
I/flutter ( 7514): Loading locale: en
I/flutter ( 7514): UiWrapper, child: Navigator-[LabeledGlobalKey<NavigatorState>#73d8e]
I/flutter ( 7514): 14:79 ReduxLogging/INFO: {Action: Instance of 'AppRoutingAction', State: StoreState, ts: 2021-03-04 16:49:14.798812}
I/flutter ( 7514): 14:80 ReduxLogging/INFO: {Action: Instance of 'AppRoutingAction', State: StoreState, ts: 2021-03-04 16:49:14.800235}
I/flutter ( 7514): 15:12 Firebase/FINE: Initalising remote config...
D/FRCPlugin( 7514): Sending fetchTimeout: 60
D/AndroidRuntime( 7514): Shutting down VM
E/AndroidRuntime( 7514): FATAL EXCEPTION: main
E/AndroidRuntime( 7514): Process: com.rocketware.onescene, PID: 7514
E/AndroidRuntime( 7514): java.lang.IllegalArgumentException: Unsupported value: com.google.firebase.remoteconfig.internal.FirebaseRemoteConfigInfoImpl@39b686
E/AndroidRuntime( 7514): 	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:276)
E/AndroidRuntime( 7514): 	at io.flutter.plugin.common.StandardMethodCodec.encodeSuccessEnvelope(StandardMethodCodec.java:59)
E/AndroidRuntime( 7514): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:238)
E/AndroidRuntime( 7514): 	at io.flutter.plugins.firebase.firebaseremoteconfig.FirebaseRemoteConfigPlugin.lambda$onMethodCall$2(FirebaseRemoteConfigPlugin.java:167)
E/AndroidRuntime( 7514): 	at io.flutter.plugins.firebase.firebaseremoteconfig.-$$Lambda$FirebaseRemoteConfigPlugin$G3sJbc89zgSFW2R5fKq_9HoTj0c.onComplete(Unknown Source:2)
E/AndroidRuntime( 7514): 	at com.google.android.gms.tasks.zzi.run(com.google.android.gms:play-services-tasks@@17.0.2:4)
E/AndroidRuntime( 7514): 	at android.os.Handler.handleCallback(Handler.java:938)
E/AndroidRuntime( 7514): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 7514): 	at com.google.android.gms.internal.tasks.zzb.dispatchMessage(com.google.android.gms:play-services-tasks@@17.0.2:6)
E/AndroidRuntime( 7514): 	at android.os.Looper.loop(Looper.java:223)
E/AndroidRuntime( 7514): 	at android.app.ActivityThread.main(ActivityThread.java:7660)
E/AndroidRuntime( 7514): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 7514): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
E/AndroidRuntime( 7514): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
I/Process ( 7514): Sending signal. PID: 7514 SIG: 9
Lost connection to device.
Exited (sigterm)

```

</details>

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[√] Flutter (Channel stable, 2.0.0, on Microsoft Windows [Version 10.0.19042.804],
    locale en-GB)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)     
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more      details.
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code, 64-bit edition (version 1.52.1)
[√] Connected device (4 available)

! Doctor found issues in 1 category.
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.12.0
Flutter SDK 2.0.0
onesceneapp 1.0.0+1

dependencies:
- badges 1.1.6 [flutter]
- bitmap 0.0.6 [flutter ffi]
- blurhash_dart 0.2.3
- chroma 0.1.0 [vector_math logging image]
- cloud_firestore 1.0.0 [cloud_firestore_platform_interface cloud_firestore_web firebase_core firebase_core_platform_interface flutter meta]
- cloud_firestore_platform_interface 4.0.0 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_functions 1.0.0 [cloud_functions_platform_interface cloud_functions_web firebase_core firebase_core_platform_interface flutter]
- confetti 0.5.4+1 [flutter random_color vector_math]
- cupertino_icons 0.1.3
- device_info 2.0.0 [flutter device_info_platform_interface]
- dotted_border 1.0.7 [flutter path_drawing]
- emoji_picker 0.1.0 [flutter shared_preferences]
- encrypt 3.2.0 [asn1lib collection pointycastle args]
- equatable 1.2.6 [collection meta]
- exif 0.1.5 [args collection convert sprintf]
- firebase_analytics 5.0.15 [meta flutter firebase_analytics_platform_interface]     
- firebase_auth 1.0.0 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.0.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_messaging 9.0.0 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_remote_config 0.9.0-dev.0 [flutter firebase_core firebase_core_platform_interface firebase_remote_config_platform_interface]
- firebase_storage 8.0.0 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]       
- flutter_image 3.0.0 [flutter]
- flutter_localizations 0.0.0 [flutter intl characters clock collection meta path typed_data vector_math]
- flutter_page_transition 0.1.6 [flutter]
- flutter_redux 0.7.0 [meta redux flutter]
- flutter_redux_navigation 0.6.0 [flutter redux]
- geolocation 1.1.2 [flutter streams_channel]
- get_ip 0.4.0 [flutter]
- google_sign_in 4.5.9 [google_sign_in_platform_interface flutter meta google_sign_in_web]
- googleapis 1.0.0 [_discoveryapis_commons http]
- googleapis_auth 1.0.0 [crypto http]
- great_circle_distance 1.0.1
- image 3.0.1 [archive meta xml]
- image_cropper 1.3.1 [flutter]
- image_picker 0.6.5+3 [flutter flutter_plugin_android_lifecycle]
- in_app_purchase 0.3.5+1 [async collection flutter json_annotation meta]
- intl 0.17.0 [clock path]
- logging 0.11.4
- mdi 3.0.0 [flutter]
- meta 1.3.0
- onescene_brand 1.0.0 [flutter os_core]
- os_core 0.0.1 [redux redux_logging quiver_hashcode intl uuid http encrypt key_value_store built_value built_collection quiver archive recase]
- package_info 2.0.0 [flutter]
- palette_generator 0.2.3 [flutter collection path]
- path_provider 1.6.1 [flutter platform meta]
- quiver 2.1.5 [matcher meta]
- recase 3.0.1
- redux 4.0.0+3
- redux_logging 0.4.0 [redux logging]
- reorderables 0.3.2 [flutter]
- settings_ui 0.6.0 [flutter]
- shared_preferences 0.5.7+3 [meta flutter shared_preferences_platform_interface shared_preferences_macos shared_preferences_web]
- uuid 3.0.1 [crypto]
- video_player 0.10.12+5 [meta video_player_platform_interface video_player_web flutter]

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math 
async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- _discoveryapis_commons 0.2.1 [http]
- archive 3.1.2 [crypto path]
- args 1.6.0
- asn1lib 0.5.15
- async 2.5.0 [collection]
- boolean_selector 2.1.0 [source_span string_scanner]
- built_collection 5.0.0
- built_value 8.0.0 [built_collection collection fixnum]
- characters 1.1.0
- charcode 1.2.0
- clock 1.1.0
- cloud_firestore_web 1.0.0 [cloud_firestore_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- cloud_functions_platform_interface 5.0.0 [firebase_core flutter meta plugin_platform_interface]
- cloud_functions_web 4.0.0 [cloud_functions_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.15.0
- convert 2.1.1 [charcode typed_data]
- crypto 3.0.0 [collection typed_data]
- device_info_platform_interface 2.0.1 [flutter meta plugin_platform_interface]      
- fake_async 1.2.0 [clock collection]
- ffi 0.1.3
- firebase_analytics_platform_interface 1.1.0 [flutter meta]
- firebase_auth_platform_interface 4.0.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 1.0.0 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.0.0 [flutter meta plugin_platform_interface]    
- firebase_core_web 1.0.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_messaging_platform_interface 2.0.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 1.0.0 [firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins js meta]
- firebase_remote_config_platform_interface 0.2.0-dev.0 [flutter meta firebase_core plugin_platform_interface]
- firebase_storage_platform_interface 2.0.0 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 1.0.0 [async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http js meta]
- fixnum 1.0.0
- flutter_plugin_android_lifecycle 1.0.11 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- google_sign_in_platform_interface 1.1.2 [flutter meta quiver]
- google_sign_in_web 0.9.2 [google_sign_in_platform_interface flutter flutter_web_plugins meta js]
- http 0.13.0 [http_parser meta path pedantic]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]      
- js 0.6.3
- json_annotation 3.1.1
- key_value_store 1.0.0
- matcher 0.12.10 [stack_trace]
- path 1.8.0
- path_drawing 0.4.1+1 [vector_math meta path_parsing flutter]
- path_parsing 0.1.4 [vector_math meta]
- pedantic 1.11.0
- petitparser 4.0.2 [meta]
- platform 2.2.1
- plugin_platform_interface 2.0.0 [meta]
- pointycastle 1.0.2
- quiver_hashcode 2.0.0
- random_color 1.0.5 [flutter]
- shared_preferences_macos 0.0.1+11 [shared_preferences_platform_interface flutter]  
- shared_preferences_platform_interface 1.0.4 [meta flutter]
- shared_preferences_web 0.1.2+7 [shared_preferences_platform_interface flutter flutter_web_plugins meta]
- sky_engine 0.0.99
- source_span 1.8.0 [charcode collection path term_glyph]
- sprintf 4.1.0
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- streams_channel 0.3.0 [flutter]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.2.19 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0 [collection]
- vector_math 2.1.0
- video_player_platform_interface 2.2.0 [flutter meta]
- video_player_web 0.1.4+1 [flutter flutter_web_plugins meta video_player_platform_interface]
- xml 5.0.2 [collection meta petitparser]
```

</details>

---

Thanks

### PR DESCRIPTION
**Pull request by Salakar**
## Description

Currently on Android the task returned by `RemoteConfig.ensureInitialized()` returns a `FirebaseRemoteConfigInfo` instance which gets sent to the Flutter standard message codec and causes a crash as this isn't a supported value that can be serialized by the codec. 

As we don't at this point care about the value of the task result, I've wrapped the task in a [`whenAll`](https://developers.google.com/android/reference/com/google/android/gms/tasks/Tasks#whenAll(com.google.android.gms.tasks.Task%3C?%3E...)) which will just return void when completed.

## Related Issues

Fixes #5222 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java [hunk 1]
```diff
@@ -132,8 +133,8 @@ public void onMethodCall(MethodCall call, final MethodChannel.Result result) {
         }
       case "RemoteConfig#setConfigSettings":
         {
-          int fetchTimeout = call.argument("fetchTimeout");
-          int minimumFetchInterval = call.argument("minimumFetchInterval");
+          int fetchTimeout = Objects.requireNonNull(call.argument("fetchTimeout"));
+          int minimumFetchInterval = Objects.requireNonNull(call.argument("minimumFetchInterval"));
```
Comment (Salakar) [inline packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java:137]: Never null as these come from Dart where they are required values.

File: packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java [hunk 2]
```diff
@@ -144,7 +145,7 @@ public void onMethodCall(MethodCall call, final MethodChannel.Result result) {
         }
       case "RemoteConfig#setDefaults":
         {
-          Map<String, Object> defaults = call.argument("defaults");
+          Map<String, Object> defaults = Objects.requireNonNull(call.argument("defaults"));
```
Comment (Salakar) [inline packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java:148]: Never null as this comes from Dart where it is a required value.

File: packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java [hunk 3]
```diff
@@ -203,25 +204,23 @@ private String mapLastFetchStatus(int status) {
...
       case FirebaseRemoteConfig.LAST_FETCH_STATUS_SUCCESS:
         return "success";
-      case FirebaseRemoteConfig.LAST_FETCH_STATUS_FAILURE:
-        return "failure";
       case FirebaseRemoteConfig.LAST_FETCH_STATUS_THROTTLED:
         return "throttled";
       case FirebaseRemoteConfig.LAST_FETCH_STATUS_NO_FETCH_YET:
         return "noFetchYet";
+      case FirebaseRemoteConfig.LAST_FETCH_STATUS_FAILURE:
       default:
```
Comment (Salakar) [inline packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java:212]: Duplicate return cases, default and failure return the same thing

File: packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java [hunk 4]
```diff
@@ -203,25 +204,23 @@ private String mapLastFetchStatus(int status) {
...
 
   private String mapValueSource(int source) {
     switch (source) {
-      case FirebaseRemoteConfig.VALUE_SOURCE_STATIC:
-        return "static";
       case FirebaseRemoteConfig.VALUE_SOURCE_DEFAULT:
         return "default";
       case FirebaseRemoteConfig.VALUE_SOURCE_REMOTE:
         return "remote";
+      case FirebaseRemoteConfig.VALUE_SOURCE_STATIC:
```
Comment (Salakar) [inline packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java:223]: Duplicate return cases, default and static return the same thing

File: packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java [hunk 5]
```diff
@@ -56,8 +58,7 @@ public void onDetachedFromEngine(FlutterPluginBinding binding) {
         cachedThreadPool,
         () -> {
           Map<String, Object> configProperties = getConfigProperties(remoteConfig);
-          Map<String, Object> configValues = new HashMap<>();
-          configValues.putAll(configProperties);
+          Map<String, Object> configValues = new HashMap<>(configProperties);
```
Comment (Salakar) [inline packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java:61]: Passing the initial values direct to the constructor. `putAll` unnecessary in this case.

File: packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java [hunk 6]
```diff
@@ -100,14 +101,14 @@ private FirebaseRemoteConfig getRemoteConfig(Map<String, Object> arguments) {
...
-  public void onMethodCall(MethodCall call, final MethodChannel.Result result) {
+  public void onMethodCall(MethodCall call, @NonNull final MethodChannel.Result result) {
     Task<?> methodCallTask;
     FirebaseRemoteConfig remoteConfig = getRemoteConfig(call.arguments());
 
     switch (call.method) {
       case "RemoteConfig#ensureInitialized":
         {
-          methodCallTask = remoteConfig.ensureInitialized();
+          methodCallTask = Tasks.whenAll(remoteConfig.ensureInitialized());
```
Comment (Salakar) [inline packages/firebase_remote_config/firebase_remote_config/android/src/main/java/io/flutter/plugins/firebase/firebaseremoteconfig/FirebaseRemoteConfigPlugin.java:111]: Currently on Android the task returned by `RemoteConfig.ensureInitialized()` returns a FirebaseRemoteConfigInfo instance which gets sent to the Flutter standard message codec and causes a crash as this isn't a supported value that can be serialized by the codec.

As we don't at this point care about the value of the task result, I've wrapped the task in a [`whenAll`](https://developers.google.com/android/reference/com/google/android/gms/tasks/Tasks#whenAll(com.google.android.gms.tasks.Task%3C?%3E...)) which will just return `void` when completed.

File: packages/firebase_remote_config/firebase_remote_config/example/test_driver/firebase_remote_config_e2e.dart [hunk 7]
```diff
@@ -22,6 +22,7 @@ void main() {
         'welcome': 'default welcome',
         'hello': 'default hello',
       });
+      await remoteConfig.ensureInitialized();
```
Comment (Salakar) [inline packages/firebase_remote_config/firebase_remote_config/example/test_driver/firebase_remote_config_e2e.dart:25]: Added a call to the previously crashing method to the tests setup - this is why we haven't seen the crash on tests.

**Review by russellwheatley — APPROVED** [review]
Provided it passes CI, LGTM.

**Pull request merged**